src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```
@@ -102,11 +104,45 @@ def validate_query_fields(namespace):
         namespace.fields = fields
 
+
 def validate_feature_query_fields(namespace):
     if namespace.fields:
         fields = []
         for field in namespace.fields:
             for feature_query_field in FeatureQueryFields:
                 if field.lower() == feature_query_field.name.lower():
                     fields.append(feature_query_field)
-        namespace.fields = fields
\ No newline at end of file
+        namespace.fields = fields
+
+
+def validate_filter_parameters(namespace):
+    """ Extracts multiple space-separated filter paramters in name[=value] format """
+    if isinstance(namespace.filterParameters, list):
+        filter_parameters_dict = {}
+        for item in namespace.filterParameters:
+            param_tuple = validate_filter_parameter(item)
+            if param_tuple:
+                param_name, param_value = param_tuple
+                # If param_name already exists, convert the values to a list
+                if param_name in filter_parameters_dict:
+                    old_param_value = filter_parameters_dict[param_name]
+                    if isinstance(old_param_value, list):
+                        old_param_value.append(param_value)
+                    else:
+                        filter_parameters_dict[param_name] = [old_param_value, param_value]
+                else:
+                    filter_parameters_dict.update({param_name: param_value})
+        namespace.filterParameters = filter_parameters_dict
+
+
+def validate_filter_parameter(string):
+    """ Extracts a single filter parameter in name[=value] format """
+    result = ()
+    if string:
+        comps = string.split('=', 1)
+        # Ignore invalid arguments like  '=value' or '='
+        if comps[0]:
+            result = (comps[0], comps[1]) if len(comps) > 1 else (string, '')
+        else:
+            logger.warning("Ignoring filter parameter '%s' because parameter name is empty.", string)
+    return result
```

if parameter key is null, we should stop and error out instead of continuing.

---

Mirroring the behavior from portal, I thought  we could just ignore the parameters with empty name and continue with the rest of changes just like portal allows users to do it.

---
In reply to: [332228746](https://github.com/avanigupta/azure-cli/pull/2#discussion_r332228746) [](ancestors = 332228746)

---

considering we will log warning. it is fine to ignore it. 

---
In reply to: [332276635](https://github.com/avanigupta/azure-cli/pull/2#discussion_r332276635) [](ancestors = 332276635,332228746)